pkg/engine/engine.go

```diff
@@ -18,22 +18,33 @@ import (
 	"context"
 	"fmt"
 	"sort"
+	"time"
 
 	"github.com/snyk/policy-engine/pkg/bundle"
 	"github.com/snyk/policy-engine/pkg/bundle/base"
 	"github.com/snyk/policy-engine/pkg/data"
+	"github.com/snyk/policy-engine/pkg/internal/withtimeout"
 	"github.com/snyk/policy-engine/pkg/logging"
 	"github.com/snyk/policy-engine/pkg/metrics"
 	"github.com/snyk/policy-engine/pkg/models"
 	"github.com/snyk/policy-engine/pkg/policy"
 )
 
+const (
+	DefaultInitTimeout  = 30 * time.Second
+	DefaultEvalTimeout  = 5 * time.Minute
+	DefaultQueryTimeout = 45 * time.Second
+)
+
 // Engine is responsible for evaluating some States with a given set of rules.
 type Engine struct {
 	// InitializationErrors contains any errors that occurred during initialization.
 	InitializationErrors []error
 	instrumentation      *engineInstrumentation
 	policySets           []*policySet
+	initTimeout          time.Duration
+	evalTimeout          time.Duration
+	queryTimeout         time.Duration
```

Comment:
What do you think of grouping these in a `type Timeouts struct`?

Reply:
I like that! Done in f42009c